src/main/java/com/upplication/s3fs/S3FileSystemProvider.java

```diff
@@ -595,6 +635,30 @@ public Properties loadAmazonProperties() {
         return props;
     }
 
+    private Properties getBucketProfileMapping() {
+        if (bucketProfileMapping != null) {
+            return bucketProfileMapping;
+        }
```

Comment:
It looks like you are trying to cache the profile mappings to avoid having to read the file on every request, but I don't see where you are ever setting bucketProfileMapping. Did I miss it?